### ISSUE TITLE
Se puede usar Vulkan?

### ISSUE THREAD

**Issue by Jbat1Jumper** [mlrdevgit/eci-2018#2]
Le quiero meter mano a esta biblioteca en particular https://github.com/vulkano-rs/vulkano
Probando y jugando con el ejemplo de [acá](https://github.com/vulkano-rs/vulkano-examples/blob/master/triangle/main.rs) compila y corre en mac. Por ahora no tengo un windows con placa de video de este siglo para probar, pero debería andar sin mucho problema.

**Comment by mlrdevgit** [comment]
Sí, podés usar esa biblioteca (aunque te pido que el proyecto pueda compilarse y correr en Windows - el nivel de hardware no calienta para la escena que hay que componer).

Pero ojo que es un poco más complicado usar Vulkan que algo tipo OpenGL clásico o DX (pre-12). Profundizo (y, por cierto, explico un poco más cómo funciona esta parte).

En particular, cuando corran shaders, van a darle recursos (vimos por ejemplo cómo pasar matrices en un 'constant buffer' durante la clase). Esos buffers vienen configurados como resource views de algún tipo, y los asignan por tipo de "slot" y número en el shader (por ejemplo, `Texture2D<float4> t : register(t0);` para la primer textura), y luego a la hora de preparar trabajo para el GPU, indicás el recurso con algo tipo [PSSetShaderResources](https://docs.microsoft.com/en-us/windows/desktop/api/d3d11/nf-d3d11-id3d11devicecontext-pssetshaderresources).

Ahora, una de las complicaciones de D3D12 y Vulkan es que estos resource views (o descriptors en Vulkan) están más bajo el control de la aplicación, con lo cual tienen que manejar a mano cuál es el layout en memoria de los mismos (en vez de pasarse simplemente por un API). En D3D12 esto se maneja con descriptor heaps, en Vulkan se maneja como descriptor sets. No es que sea particularmente difícil, sobre todo para algo sencillo es bastante mecánico, pero fijate por ejemplo las notas en https://github.com/vulkano-rs/vulkano/blob/master/vulkano/src/descriptor/mod.rs para ver qué necesitás en la librería que mencionaste.
